src/coreComponents/dataRepository/GroupContext.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file GroupContext.cpp
+ */
+
+#include "GroupContext.hpp"
+
+namespace geos
+{
+namespace dataRepository
+{
+
+
+GroupContext::GroupContext( Group & group, string const & objectName ):
+  DataContext( objectName ),
+  m_group( group )
+{}
+GroupContext::GroupContext( Group & group ):
+  GroupContext( group, group.getName() )
+{}
+
+string GroupContext::toString() const
+{
+  std::vector< ToStringInfo > parentsInfo;
+  for( Group const * group = &m_group; group->hasParent(); group = &group->getParent() )
+  {
+    parentsInfo.push_back( group->getDataContext().getToStringInfo() );
+  }
+
+  std::ostringstream path;
+  auto lastFileInfo = &*std::find_if( parentsInfo.begin(), parentsInfo.end(),
+                                      []( ToStringInfo const & i ) { return i.hasInputFileInfo(); } );
+  for( auto info = parentsInfo.rbegin(); info != parentsInfo.rend(); ++info )
+  {
+    path << ( &*info != lastFileInfo ?
```

Comment:
Can't you simply compare `lastFileInfo` and `info` without the `&*` here and there?
Or at least
```suggestion
  auto lastFileInfo = std::find_if( parentsInfo.begin(), parentsInfo.end(),
                                    []( ToStringInfo const & i ) { return i.hasInputFileInfo(); } );
  for( auto info = parentsInfo.rbegin(); info != parentsInfo.rend(); ++info )
  {
    path << ( &*info != &*lastFileInfo ?
```
?

Reply:
Does the `info.base() != lastFileInfo` syntax would be better ?

Reply:
My bad, this does not give the right result, we would need `( info.base() - 1 ) != lastFileInfo` or `std::prev( info.base() ) != lastFileInfo` here.
To avoid this confusing pointer arithmetic, we should stick to `&*info != &*lastFileInfo`.
Another solutions would be to reverse `parentsInfo` rather than using reverse iterators, or to use `rbegin()` and `rend()` as `std::find_if()` parameters.

Reply:
I'm fine with your code if it's working. This `&*` is a bit unusual, but I guess it is what it is.

Reply:
Dereferencing `lastinfo` is undefined behavior if the `find_if` call doesn't find anything. I wouldn't mind `std::prev( info.base() ) != lastFileInfo`, maybe you can add a short comment on the same line to help the reader, something like `// pointing to the same element`.

Reply:
Oh, I should have seen that, thank you for checking!